tests/postgres_tests/test_indexes.py

```diff
@@ -172,6 +176,35 @@ def test_gin_fastupdate(self):
             editor.remove_index(IntegerArrayModel, index)
         self.assertNotIn(index_name, self.get_constraints(IntegerArrayModel._meta.db_table))
 
+    def test_partial_gin_index(self):
+        with register_lookup(CharField, Length):
+            index_name = 'char_field_gin_partial_idx'
+            index = GinIndex(fields=['field'], name=index_name, condition=Q(field__length=40))
+            with connection.schema_editor() as editor:
+                editor.add_index(CharFieldModel, index)
+            constraints = self.get_constraints(CharFieldModel._meta.db_table)
+            self.assertEqual(constraints[index_name]['type'], 'gin')
+            with connection.schema_editor() as editor:
+                editor.remove_index(CharFieldModel, index)
+            self.assertNotIn(index_name, self.get_constraints(CharFieldModel._meta.db_table))
+
+    def test_partial_gin_index_with_tablespace(self):
+        with register_lookup(CharField, Length):
+            index_name = 'char_field_gin_partial_idx'
+            index = GinIndex(
+                fields=['field'],
+                name=index_name,
+                condition=Q(field__length=40),
+                db_tablespace='pg_default',
+            )
+            with connection.schema_editor() as editor:
+                editor.add_index(CharFieldModel, index)
+            constraints = self.get_constraints(CharFieldModel._meta.db_table)
+            self.assertEqual(constraints[index_name]['type'], 'gin')
+            with connection.schema_editor() as editor:
+                editor.remove_index(CharFieldModel, index)
+            self.assertNotIn(index_name, self.get_constraints(CharFieldModel._meta.db_table))
```

Comment:
Do we want to assert something about the `tablespace` here?

Reply:
I see that I just asserted that `TABLESPACE` would be part of the generated SQL. Maybe that's also acceptable, too.